cli/src/pcluster/resources/imagebuilder/parallelcluster.yaml

```diff
@@ -213,6 +213,11 @@ phases:
               OS='{{ build.OperatingSystemName.outputs.stdout }}'
               PLATFORM='{{ build.PlatformName.outputs.stdout }}'
 
+              if [ ${CfnParamUpdateOsAndReboot} == false ] && [ ${!PLATFORM} == DEBIAN ]; then
```

Comment:
we change the
`elif [[ ${!OS} =~ ^ubuntu(18|20)04$ ]]; then`
to
` elif [[ ${!PLATFORM} == DEBIAN ]]; then`
and move the two commands in there

i.e.
```
              elif [[ ${!PLATFORM} == DEBIAN ]]; then
                   flock $(apt-config shell StateDir Dir::State/d | sed -r "s/.*'(.*)\/?'$/\1/")/daily_lock systemctl stop apt-daily.timer apt-daily.service apt-daily-upgrade.timer apt-daily-upgrade.service
                DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" --with-new-pkgs upgrade && apt-get autoremove -y         
                  apt-cache search build-essential
                  apt-get clean
                  apt update -y 
              fi
```

Reply:
done

Reply:
update is missing the `-y`